extensions/scarb-execute/src/profiler.rs

```diff
@@ -0,0 +1,109 @@
+use crate::output::ExecutionResources;
+use anyhow::{Result, anyhow};
+use cairo_annotations::trace_data::{
+    CairoExecutionInfo, CallEntryPoint as ProfilerCallEntryPoint, CallTraceV1 as ProfilerCallTrace,
+    CallType as ProfilerCallType, CasmLevelInfo, EntryPointType as ProfilerEntryPointType,
+    ExecutionResources as ProfilerExecutionResources, TraceEntry as ProfilerTraceEntry,
+};
+use cairo_annotations::trace_data::{ContractAddress, EntryPointSelector};
+use cairo_vm::vm::trace::trace_entry::RelocatedTraceEntry;
+use camino::Utf8PathBuf;
+use scarb_extensions_cli::execute::ExecutionTarget;
+use scarb_metadata::PackageMetadata;
+
+#[derive(Default, Debug)] // todo: add to docs somewhere
+pub enum TrackedResource {
+    #[default]
+    CairoSteps,
+    SierraGas,
+}
+
+impl From<&str> for TrackedResource {
+    fn from(m: &str) -> Self {
+        match m {
+            "sierra-gas" => TrackedResource::SierraGas,
+            _ => TrackedResource::CairoSteps,
+        }
+    }
+}
+
+impl From<TrackedResource> for &str {
+    fn from(r: TrackedResource) -> Self {
+        match r {
+            TrackedResource::SierraGas => "sierra-gas",
+            TrackedResource::CairoSteps => "cairo-steps",
+        }
+    }
+}
+
+pub fn build_profiler_call_trace(
+    target: &ExecutionTarget,
+    vm_trace: Option<Vec<RelocatedTraceEntry>>,
+    vm_resources: ExecutionResources,
+    tracked_resource: &TrackedResource,
+    source_sierra_path: Utf8PathBuf,
+) -> Result<ProfilerCallTrace> {
+    let entry_point = build_profiler_call_entry_point(target);
+    let profiler_vm_trace = vm_trace
+        .as_ref()
+        .map(|trace_data| trace_data.iter().map(build_profiler_trace_entry).collect())
+        .unwrap();
+    let cairo_execution_info = CairoExecutionInfo {
+        casm_level_info: CasmLevelInfo {
+            run_with_call_header: false,
+            vm_trace: profiler_vm_trace,
+        },
+        source_sierra_path,
+    };
+
+    let mut execution_resources: ProfilerExecutionResources =
+        vm_resources.try_into().map_err(|e: String| anyhow!(e))?;
+
+    // currently (CallTraceV1) cairo-profiler uses vm_resources and gas_consumed to determine what was tracked
+    // so we need to reset a value for specific tracked resource, to not confuse it
```

Comment:
NIT
```suggestion
    // The `CallTraceV1` cairo-profiler trace format uses `vm_resources` and `gas_consumed` to determine what was tracked. 
    // Exactly one of the two should be defined - indicating the resource that has been tracked, while the other should be reset.
    // Since we cannot infer the tracked resources from execution directly, we mask one of the fields based on user provided configuration.
    // This maintains compatibility with other execution environments that produce traces in `CallTraceV1` format.
```